### PR TITLE
Refactor login flow types into matrix-js-sdk

### PR DESCRIPTION
**Pull request by hughns**
The corresponding PR for matrix-js-sdk is https://github.com/matrix-org/matrix-js-sdk/pull/2669 which must be merged at the same time (was originally part of ~~https://github.com/matrix-org/matrix-js-sdk/pull/2633~~)

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by turt2live — CHANGES_REQUESTED** [review]
it appears as though a commit is missing - this does not compile.

**Comment by hughns** [comment]
@turt2live unless you are looking at something different, is the issue that it depends on the js-sdk branch given by https://github.com/matrix-org/matrix-js-sdk/pull/2633?

**Comment by hughns** [comment]
@turt2live the required changes are now on js-sdk develop branch but don't seem to be pulled in for the test on this PR. What have I done wrong? 🤦‍♂️

**Comment by turt2live** [comment]
it's probably the lockfile being a huge pain, but let's see if another merge does it.

Initial review was based on the diff being a large shade of red, fwiw - didn't see the js-sdk PR at all and it should have considered it as part of the build.

**Comment by turt2live** [comment]
oh, the react-sdk PR got reverted.

@hughns what's the status of this?

**Comment by hughns** [comment]
@turt2live the status is that there is a new smaller PR https://github.com/matrix-org/matrix-js-sdk/pull/2669 which does the remaining change (which is what broke react-sdk earlier in the week).

File: src/Login.ts [hunk 1]
```diff
@@ -19,54 +19,19 @@ limitations under the License.
 import { createClient } from "matrix-js-sdk/src/matrix";
 import { MatrixClient } from "matrix-js-sdk/src/client";
 import { logger } from "matrix-js-sdk/src/logger";
+import { ILoginParams, LoginFlow } from "matrix-js-sdk/src/@types/auth";
 
 import { IMatrixClientCreds } from "./MatrixClientPeg";
 import SecurityCustomisations from "./customisations/Security";
 
+export {
+    IdentityProviderBrand, IIdentityProvider, ISSOFlow, LoginFlow,
```
Comment (turt2live) [inline src/Login.ts:28]:
```suggestion
    IdentityProviderBrand,
    IIdentityProvider,
    ISSOFlow,
    LoginFlow,
```

**Review by turt2live — APPROVED** [review]

**Pull request merged**